openedx/core/djangoapps/user_api/management/tests/test_bulk_rehash_retired_usernames.py

```diff
@@ -0,0 +1,66 @@
+"""
+Test the bulk_rehash_retired_usernames management command
+"""
+import pytest
+
+from django.core.management import call_command
+
+from openedx.core.djangoapps.user_api.accounts.tests.retirement_helpers import RetirementTestCase, fake_retirement
+from openedx.core.djangoapps.user_api.models import UserRetirementStatus
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+from student.tests.factories import UserFactory
+
+pytestmark = pytest.mark.django_db
+
+
+def _setup_users():
+    # When we loop through creating users, take additional action on these
+    user_indexes_to_be_fake_retired = (2, 4, 6, 8, 10)
+    user_indexes_to_be_rehashed = (4, 6)
+
+    users_skipped = []
+    users_faked = []
+    users_needing_rehash = []
+
+    # Create some test users with retirements
+    for i in range(1, 11):
+        user = UserFactory()
+        retirement = UserRetirementStatus.create_retirement(user)
+
+        if i in user_indexes_to_be_fake_retired:
+            fake_retirement(user)
+
+            if i in user_indexes_to_be_rehashed:
+                # In order to need a rehash user.username must be the same as
+                # retirement.retired_username and NOT the same as the hash
+                # generated when the script is run. So we force that here.
+                retirement.retired_username = 'bogus {}'.format(i)
```

Comment:
Nit: Would this condition describe the original error better?:
`retirement.retired_username = retirement.retired_username.upper()`
I don't think it's of utmost importance, though...